isisd/isis_circuit.c

```diff
@@ -1446,18 +1450,23 @@ int isis_if_delete_hook(struct interface *ifp)
 	/* Clean up the circuit data */
 	if (ifp && ifp->info) {
 		circuit = ifp->info;
-		isis_csm_state_change(IF_DOWN_FROM_Z, circuit, circuit->area);
+		isis_csm_state_change(IF_DOWN_FROM_Z, circuit, ifp);
 	}
 
 	return 0;
 }
 
 static int isis_ifp_create(struct interface *ifp)
 {
-	if (if_is_operative(ifp))
+	struct vrf *vrf = NULL;
+
+	if (if_is_operative(ifp)) {
```

Comment:
why does the interface need to be operative?  What happens if I am pre-creating config under an interface that does not exist yet?

Reply:
This check is part of existing code. 
The interface needs to be operative to create the circuit. 
If you are pre-creating config under an interface that does not exist yet, then the circuit will not be created. When zebra sends an event for interface UP & RUNNING (IF_UP_FROM_Z)  then the circuit will be created.